core/unit_test/CMakeLists.txt

```diff
@@ -329,6 +329,30 @@ foreach(PairDeviceSpace HIP-HostPinned;HIP-Managed;Cuda-HostPinned;Cuda-UVM;SYCL
   endif()
 endforeach()
 
+foreach(PairDeviceSpace HIP-Managed;Cuda-UVM;SYCL-SharedUSM)
+  string(REGEX REPLACE "([^-]*)-(.*)" "\\1" DEVICE ${PairDeviceSpace})
+  string(REGEX REPLACE "([^-]*)-(.*)" "\\2" SPACE ${PairDeviceSpace})
+
+  string(TOUPPER ${DEVICE} UPPER_DEVICE)
+  string(TOLOWER ${DEVICE} dir)
+
+  if(Kokkos_ENABLE_${UPPER_DEVICE})
+    set(dir ${CMAKE_CURRENT_BINARY_DIR}/${dir})
+    file(MAKE_DIRECTORY ${dir})
+    set (Name PageMigration)
+    set(file ${dir}/Test${DEVICE}${SPACE}_${Name}.cpp)
+    # Write to a temporary intermediate file and call configure_file to avoid
+    # updating timestamps triggering unnecessary rebuilds on subsequent cmake runs.
+    file(WRITE ${dir}/dummy.cpp
+        "#include <Test${DEVICE}${SPACE}_Category.hpp>\n"
+        "#include <Test${Name}.hpp>\n"
+    )
+    configure_file(${dir}/dummy.cpp ${file})
+    list(APPEND ${DEVICE}_SOURCES4 ${file})
+    list(APPEND ${DEVICE}_SOURCES ${${DEVICE}_SOURCES4})
```

Comment:
`SYCL` doesn't use `${DEVICE}_SOURCES` (since running the device compiler and linking all tests at once is a nightmare) so you would need to define an additional test target for `SYCL` that uses `${DEVICE}_SOURCES4`.

Reply:
done in [8ffb58](https://github.com/kokkos/kokkos/pull/5289/commits/8ffb58b4aaf156f79b08cfed392cf3387a210145)